src/app/shared/forge.service.ts

```diff
@@ -20,14 +21,24 @@ export class ForgeService {
     this.apiUrl += "launchpad";
   }
 
+  private options(): any {
+    var headers = new Headers();
+    if( this.filters ) {
+      headers.append("LAUNCHPAD_BACKEND_LABEL_FILTERS", this.filters);
```

Comment:
Better have a `X-` preffix to comply with standard HTTP additional headers

Reply:
that's not a bad idea, but it's not what the LaunchResource currently expects.  See:

https://github.com/openshiftio/launchpad-backend/blob/master/src/main/java/io/openshift/launchpad/backend/rest/LaunchResource.java#L468

Reply:
LaunchResource adds all header entries, regardless of their name. The X- is just a convention for header entries that are not part of the HTTP spec

Reply:
will the LauchResource strip the X- prefix?

Reply:
No, you need to change the addon to use the X- prefix too

Reply:
I can do it, but that seems like a leaky abstraction since HTTP concepts are making it into the addon.

Reply:
Why are these filters are passed to the backend in the HTTP Header? Why not as an UISelectMany input?

Reply:
Are they supposed to be provided by the frontend? Or is it a parameter that is set in the backend?

Reply:
Using the HTTP header feature of LaunchResource seemed like the easiest way to pass the parameter from the UI to the addon since it's optional and I did not want it showing up in the ui forms.

Reply:
Perhaps the cleanest thing to do would be to enhance LaunchResource to strip the X- prefix on headers.

Reply:
Is the UI supposed to provide these values? It seems that they are meant to be set as an env var in the backend?

Reply:
Sorry about being too dense here, just want to make sure I understand what's the use case behind this

Reply:
Yeah. UI provides the label filter.

Reply:
This PR should allow me to add the X- prefix.

Reply:
https://github.com/openshiftio/launchpad-backend/pull/66